hooks/useLocalStorageState.js

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useState } from 'react';
+
+function useLocalStorageState(key, defaultVa1ue) {
+    const [state, setState] = useState(() => {
+        let value;
+        try {
+            value = JSON.parse(window.localStorage.getItem(key) || String(defaultVa1ue))
+        }
+        catch (e) {
+            value = defaultVa1ue;
+        }
+        return value;
+    })
+    useEffect(
+        () => {
+            window.localStorage.setItem(key, state);
+        }, [state]);
+    return [state, setState];
+}
+export default function Counter() {
```

Comment:
You need to export the hook and not the component. Also, If possible add unit tests for your code. You can take help from the already written unit tests.